### PR TITLE
feat: deposit transaction (op stack)

### PR DESCRIPTION
**Pull request by jxom**
## PR-Codex overview
This PR focuses on making various updates and improvements across different files in the codebase. 

### Detailed summary
- Updated `createEventFilter.ts` and `createContractEventFilter.ts` to include a new `strict` parameter.
- Updated `estimateContractL1Gas.test.ts` to modify the expected value in a test case.
- Added new functions `parseEventLogs` and `parseGwei` to `index.test.ts`.
- Updated `chainConfig.ts` to import and include `serializers`.
- Added new files `vite.config.ts` and `tsconfig.node.json` in the `op-stack_deposit` example.
- Updated `.env.example` to include new environment variables related to Optimism.
- Updated `index.ts` to export `parseEventLogs` function.
- Updated `chain.ts` files to include `sourceId` in the chain definitions.
- Updated `formatters.ts` to import and use new types related to block and transaction.
- Updated `waitForTransactionReceipt.ts` to handle error conditions.
- Updated `package.json` in the `op-stack_deposit` example to include new dependencies and scripts.
- Updated `formatters.test-d.ts` to import and use new types related to block and transaction.
- Updated `zora.ts`, `optimism.ts`, `base.ts`, `zoraTestnet.ts`, `optimismGoerli.ts`, `zoraSepolia.ts`, `baseGoerli.ts`, `optimismSepolia.ts` to include `sourceId` in the chain definitions.
- Updated `index.ts` to export types related to block and transaction.
- Updated `contract.ts` to include new types related to chain and target chain.
- Updated `estimateFeesPerGas.ts` to import and use new types related to chain.
- Added new files `tsconfig.json` and `tsconfig.node.json` in the `op-stack_deposit` example.
- Updated `chains.ts` to export new chain definitions.

> The following files were skipped due to too many changes: `site/docs/actions/wallet/prepareTransactionRequest.md`, `src/actions/public/waitForTransactionReceipt.test.ts`, `src/chains/definitions/baseSepolia.ts`, `src/chains/opStack/utils/getSourceHash.test.ts`, `test/src/constants.ts`, `src/clients/decorators/wallet.ts`, `src/chains/opStack/decorators/walletL1.test.ts`, `src/clients/decorators/public.ts`, `src/actions/wallet/deployContract.ts`, `src/actions/public/estimateMaxPriorityFeePerGas.ts`, `src/types/chain.ts`, `src/chains/opStack/utils/getL2TransactionHashes.ts`, `src/types/utils.ts`, `src/chains/opStack/utils/extractTransactionDepositedLogs.ts`, `src/chains/opStack/actions/estimateL1Gas.ts`, `src/chains/opStack/types/block.ts`, `test/globalSetup.ts`, `site/op-stack/utilities/fromOpaqueData.md`, `src/chains/opStack/decorators/public.test.ts`, `src/chains/opStack/actions/estimateL1Fee.test.ts`, `src/chains/opStack/actions/estimateTotalGas.test.ts`, `src/chains/opStack/actions/estimateTotalFee.test.ts`, `src/chains/opStack/utils/getL2TransactionHashes.test.ts`, `src/actions/wallet/signTransaction.ts`, `src/actions/wallet/sendTransaction.ts`, `site/op-stack/utilities/getL2TransactionHashes.md`, `src/chains/opStack/actions/estimateL1Fee.ts`, `src/actions/public/simulateContract.ts`, `src/chains/opStack/utils/fromOpaqueData.ts`, `src/chains/opStack/actions/estimateL1Gas.test.ts`, `site/op-stack/utilities/extractTransactionDepositedLogs.md`, `src/chains/opStack/utils/getL2TransactionHash.test.ts`, `site/op-stack/utilities/parseTransaction.md`, `site/op-stack/client.md`, `src/chains/opStack/utils/getL2TransactionHash.ts`, `src/actions/wallet/writeContract.ts`, `site/.vitepress/sidebar.ts`, `src/chains/opStack/utils/getSourceHash.ts`, `site/op-stack/utilities/getL2TransactionHash.md`, `src/chains/opStack/utils/fromOpaqueData.test.ts`, `src/actions/public/getFilterLogs.ts`, `src/actions/public/getLogs.ts`, `test/src/opStack.ts`, `site/op-stack/utilities/getSourceHash.md`, `src/chains/opStack/types.ts`, `src/actions/public/getFilterChanges.ts`, `site/op-stack/utilities/serializeTransaction.md`, `src/chains/opStack/serializers.ts`, `src/chains/opStack/formatters.test.ts`, `src/chains/opStack/parsers.ts`, `src/chains/opStack/decorators/walletL1.ts`, `src/chains/opStack/index.ts`, `examples/op-stack_deposit/index.tsx`, `src/chains/opStack/decorators/public.ts`, `src/chains/opStack/parsers.test.ts`, `src/utils/abi/parseEventLogs.ts`, `src/actions/wallet/prepareTransactionRequest.test.ts`, `src/chains/opStack/actions/buildDepositTransaction.ts`, `src/chains/opStack/serializers.test.ts`, `site/op-stack/actions/buildDepositTransaction.md`, `src/chains/opStack/actions/depositTransaction.ts`, `src/chains/opStack/actions/depositTransaction.test.ts`, `src/utils/abi/parseEventLogs.test.ts`, `src/chains/opStack/abis.ts`, `src/actions/wallet/prepareTransactionRequest.ts`, `site/docs/contract/parseEventLogs.md`, `site/op-stack/actions/depositTransaction.md`, `site/op-stack/guides/deposits.md`, `src/chains/opStack/actions/buildDepositTransaction.test.ts`, `src/chains/opStack/utils/extractTransactionDepositedLogs.test.ts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/chains/definitions/base.ts [hunk 1]
```diff
@@ -23,5 +25,12 @@ export const base = /*#__PURE__*/ defineChain({
...
       blockCreated: 5022,
     },
+    portal: {
+      [sourceId]: {
+        address: '0x49048044D57e1C92A77f79988d21Fa8fAF74E97e',
+        blockCreated: 17482143,
+      },
+    },
   },
+  sourceId,
```
Comment (lukasrosario) [inline src/chains/definitions/base.ts:35]: big fan of this. this is currently the messiest part of op-wagmi https://op-wagmi-docs.vercel.app/docs/configuration.html

thoughts on adding the other addresses too? L1StandardBridge etc

Comment (jxom) [reply src/chains/definitions/base.ts:35]: Yep, definitely open to that!

Comment (wilsoncusack) [reply src/chains/definitions/base.ts:35]: I think originally you had been concerned about chains object getting bloated by putting these here? Curious your thinking now. I also think I dislike the idea that contracts another chain exists in base.contracts.

Comment (jxom) [reply src/chains/definitions/base.ts:35]: > I think originally you had been concerned about chains object getting bloated by putting these here?

I was more concerned about adding contracts to the L1, which would bloat. In contrast, it is harder to bloat on the L2 chain. Realistically though, we only really add contracts here which are used by Viem, so it shouldn't be too bad.

> I also think I dislike the idea that contracts another chain exists in base.contracts.

Idea is that the `sourceId` delimiter key distinguishes this.

Comment (Songkeys) [reply src/chains/definitions/base.ts:35]: Hi. Just found this discussion. Not adding `L1StandardBridge` etc. [prevents implementing contracts like "depositErc20To"](https://github.com/wevm/viem/discussions/1646). I can help creating a PR to implment these lacking methods for op stack. But adding these contract address to every chain requires some work. Any ideas?

For reference, this is what we did in our library: https://github.com/RSS3-Network/RSS3-Explorer/blob/f253f5c8feb977d085c6668a06a8941f3aea63b6/src/lib/wagmi/config/chains.ts#L30-L71

**Pull request merged**